### ISSUE TITLE
route function

### ISSUE THREAD

**Issue by eljhfx** [duality-labs/duality#238]
takes as input an ordered list of token pairs and swaps into them consecutively

- [ ] Spec
- [ ] Implementation
- [ ] Testing

**Comment by eljhfx** [comment]
@teddyknox needed? not sure

Would it be useful and should it be a large lift: yeah it should come in handy and won't be a large life (is just defining a min price / min out for an entire route instead of a single pair)

**Comment by teddyknox** [comment]
Agreed it would be useful. Will keep it here then.

**Comment by eljhfx** [comment]
One question to ask ourselves is exactly what will this enable that wasn't previously.

Can it help us catch errors midway through routes? Will it help the UX of routing complex trades?

I know you can do some of this stuff w/ the cosmos SDK (compound txs) but not sure what is and isnt possible

**Comment by teddyknox** [comment]
@eljhfx I thought the advantage was that the best route can change between when you prepare the tx and when it executes.

**Comment by eljhfx** [comment]
Yeah but that isn't what's being specified in this issue although that'd also be useful for a function such as `dynamicRoute()` which checks a few default paths and chooses which is best (or whether some combination makes sense).
